lib/ansible/plugins/action/win_reboot.py

```diff
@@ -149,4 +144,11 @@ def run_test_command():
             result['rebooted'] = True
             result['msg'] = toex.message
 
+        if post_reboot_delay_sec != 0:
```

Comment:
I'm just thinking whether we should add in the `do_until_success_or_timeout` after the delay section in case the host is still down and we want to wait for it to come back online again.

Reply:
That's something I did not think of (because I did not need it).

My use-case was to delay a specific time after the system was ready to be used (from a WinRM perspective) to have specific stuff settle. This could be some (internal) services that take time to get ready, like AD, or post-update dust-settling. So it deliberately does not include the time it takes to reboot (which could be very different for physical hardware compared to VMs).